main.c

```diff
@@ -1401,7 +1401,10 @@ main
     }
 
     struct Mailbox *m_cur = mailbox_find(buf_string(folder));
-    mutt_folder_hook(buf_string(folder), m_cur ? m_cur->name : NULL);
+    // Take a copy of the name just in case the hook alters m_cur
+    const char *name = mutt_str_dup(m_cur ? m_cur->name : NULL);
```

Comment:
I think the wtf level would be lower if we didn't call the function for nothing :)

```suggestion
    const char *name = m_cur ? mutt_str_dup(m_cur->name) : NULL;
```